internal/monitor/monitor.go

```diff
@@ -191,21 +194,36 @@ func (pm *PowerMonitor) ensureFreshData() error {
 	return err
 }
 
+func (pm *PowerMonitor) isFresh() bool {
+	snapshot := pm.snapshot.Load()
+	if snapshot == nil || snapshot.Timestamp.IsZero() {
+		return false
+	}
+
+	age := pm.clock.Now().Sub(snapshot.Timestamp)
+	return age <= pm.maxStaleness
+}
+
 // computePower create a new snapshot of the power consumption of various levels( currently only node)
 func (pm *PowerMonitor) computePower() error {
 	started := time.Now()
 	defer func() { pm.logger.Info("Computed power", "duration", time.Since(started)) }()
 
+	prevSnapshot := pm.snapshot.Load()
+	// ensure snapshot is not nil from here on
+	if prevSnapshot == nil {
+		prevSnapshot = NewSnapshot()
+	}
```

Comment:
cant we simplify it to
```
if prevSnapshot == nil {
  	snapshot := NewSnapshot()
      	snapshot.Timestamp = time.Now()
	pm.snapshot.Store(snapshot)
        return err
}
```

otherwise both prevSnapshot and newSnapshot are same

Reply:
sorry, I am not getting it :( 
When monitor starts, we don't allocate any memory for the snapshot. In the `computePower` we ensure that previousSnapshot which is going to be referred from now on has a blank (0 ed out) snapshot. We specifically don't want to set the timestamp of the previous snapshot and it must be 0. This information is useful to compute the first - collection.

Reply:
ref: https://github.com/sustainable-computing-io/kepler/issues/2025